crest/Assets/Crest/Crest/Shaders/Resources/FFT/FFTCompute.compute

```diff
@@ -0,0 +1,161 @@
+// Crest Ocean System
+
+// Copyright 2021 Wave Harmonic Ltd
+
+// Inspired by https://github.com/speps/GX-EncinoWaves
+
+// First SIZE constant must match FFT_KERNEL_0_RESOLUTION in FFTCompute.cs
+#pragma kernel ComputeFFT SIZE=8 PASSES=3 CHANNEL=x TX=8 TY=1 FINAL=0
+#pragma kernel ComputeFFT SIZE=8 PASSES=3 CHANNEL=y TX=1 TY=8 FINAL=1
+#pragma kernel ComputeFFT SIZE=16 PASSES=4 CHANNEL=x TX=16 TY=1 FINAL=0
+#pragma kernel ComputeFFT SIZE=16 PASSES=4 CHANNEL=y TX=1 TY=16 FINAL=1
+#pragma kernel ComputeFFT SIZE=32 PASSES=5 CHANNEL=x TX=32 TY=1 FINAL=0
+#pragma kernel ComputeFFT SIZE=32 PASSES=5 CHANNEL=y TX=1 TY=32 FINAL=1
+#pragma kernel ComputeFFT SIZE=64 PASSES=6 CHANNEL=x TX=64 TY=1 FINAL=0
+#pragma kernel ComputeFFT SIZE=64 PASSES=6 CHANNEL=y TX=1 TY=64 FINAL=1
+#pragma kernel ComputeFFT SIZE=128 PASSES=7 CHANNEL=x TX=128 TY=1 FINAL=0
+#pragma kernel ComputeFFT SIZE=128 PASSES=7 CHANNEL=y TX=1 TY=128 FINAL=1
+#pragma kernel ComputeFFT SIZE=256 PASSES=8 CHANNEL=x TX=256 TY=1 FINAL=0
+#pragma kernel ComputeFFT SIZE=256 PASSES=8 CHANNEL=y TX=1 TY=256 FINAL=1
+#pragma kernel ComputeFFT SIZE=512 PASSES=9 CHANNEL=x TX=512 TY=1 FINAL=0
+#pragma kernel ComputeFFT SIZE=512 PASSES=9 CHANNEL=y TX=1 TY=512 FINAL=1
+
+// Must match CASCADE_COUNT in FFTCompute.cs
+#define CASCADE_COUNT 16
+
+float _Chop;
+
+Texture2DArray<float2> _InputH;
+Texture2DArray<float2> _InputX;
+Texture2DArray<float2> _InputZ;
+Texture2D<float2> _InputButterfly;
+#if !FINAL
+RWTexture2DArray<float2> _Output1;
+RWTexture2DArray<float2> _Output2;
+RWTexture2DArray<float2> _Output3;
+#else
+RWTexture2DArray<float3> _Output;
+#endif
+
+groupshared float2 _IntermediatesH[SIZE];
+groupshared float2 _ScratchH[SIZE];
+groupshared float2 _IntermediatesX[SIZE];
+groupshared float2 _ScratchX[SIZE];
+groupshared float2 _IntermediatesZ[SIZE];
+groupshared float2 _ScratchZ[SIZE];
+
+void ButterflyPass(float2 butterfly, uint coord, uint passIndex, uint cascade)
+{
+	uint indexA, indexB;
+
+	const uint offset = 1 << passIndex;
+	if ((coord / offset) % 2 == 1)
+	{
+		indexA = coord - offset;
+		indexB = coord;
+	}
+	else
+	{
+		indexA = coord;
+		indexB = coord + offset;
+	}
+
+	if (passIndex == 0)
+	{
+		indexA = reversebits(indexA) >> (32 - PASSES);
+		indexB = reversebits(indexB) >> (32 - PASSES);
```

Comment:
This needs the _reversebits_ fallback for it to work on older MacOS devices.

Reply:
sorry i for some reason thought it was already merged in but did not check! brain fog. added now. seems good on windows at least..